### DISCUSSION TITLE
[@types/node] v17 Blob type causing conflicts with lib.dom.ts

### DISCUSSION THREAD

**Discussion by Borewit** [DefinitelyTyped/DefinitelyTyped#58252] [Issues with a @types package]
Issue in @types/node version 17

Return type `NodeJS.ReadableStream`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/97ee04ae27d5850c7775701e2d0634f44b3ee971/types/node/v16/stream/consumers.d.ts#L11 is creating conflicts with duplicate of interface in `lib.dom.ts`

Introduced in: feat(node): v17 #57678

```ts
const stream: ReadableStream = blob.stream();
```
giving error:
```
TS2740: Type 'ReadableStream' is missing the following properties from type 'ReadableStream ': locked, cancel, getReader, pipeThroughTS2740: Type 'ReadableStream' is missing the following properties from type 'ReadableStream ': locked, cancel, getReader, pipeThrough
```

Related Dependabot PR: https://github.com/Borewit/music-metadata-browser/pull/711

[a comment by an automated account is not shown]

**Comment by Gozala** [comment]
I have proposed a fix https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59057 but it got rejected and closed after I have failed to followup on time. I might make another attempt, but I fear there is just too much context around how things are wired together for outsider like myself to succeed.

**Comment by Gozala** [comment]
@LinusU @gabritto you seem to have better ideas how to address incompatibility with lib.dom.ts could one of you take on this ? I personally have no idea how to deal with this 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/063f0c8cedb9458e4e07606fc3ba1793ff0ab705/types/node/stream/consumers.d.ts#L1-L5

And it feels like at best I'll just be making specific changes you suggest, which seems very ineffective way to pursue this.

**Comment by fredericvilcot** [comment]
Hello guys, have you some news about this issue?
I'm facing the same problem when trying to infer from `stream()` method: node types comes to override DOM lib.
@Gozala Your PR looked great, is it only because of the unit tests that your modification cannot be merged?
@nodejs Could you guys provide some help in this resolution?
Many thanks

**Reply by Gozala** [reply]
I don’t really know. The way things are tested is confusing and I could not figure out. After multiple attempts to resolve this issue I gave up
